### PR TITLE
Update BDD Markdown to v.6

### PR DESCRIPTION
**Pull request by pudkrong**
The BDD Markdown v.6 introduces breaking change.

### CODE REVIEW

File: feature-runner/run-features.ts [hunk 1]
```diff
@@ -89,33 +89,13 @@ const runner = await runFolder({
 	name: 'hello.nrfcloud.com backend',
 	logObserver: {
 		onDebug: (info, ...args) =>
-			console.error(
-				ts(),
-				chalk.magenta.dim(info.context.keyword),
-				chalk.magenta(info.context.title),
-				...args.map((arg) => chalk.cyan(print(arg))),
-			),
+			console.error(ts(), ...args.map((arg) => chalk.cyan(print(arg)))),
```
Comment (coderbyheart) [inline feature-runner/run-features.ts]: Unrelated color change. Why?

File: feature-runner/steps/config.ts [hunk 2]
```diff
@@ -18,36 +15,26 @@ const createConfigStepRunners = ({
...
 }): StepRunner<Record<string, any>>[] => {
-	const setupDeviceShadowFetchingConfiguration = async ({
-		step,
-		log: {
-			step: { progress },
-		},
-	}: StepRunnerArgs<Record<string, any>>): Promise<StepRunResult> => {
-		const match = matchGroups(
-			Type.Object({
+	const setupDeviceShadowFetchingConfiguration2 = regExpMatchedStep(
```
Comment (coderbyheart) [inline feature-runner/steps/config.ts]:
```suggestion
	const setupDeviceShadowFetchingConfiguration = regExpMatchedStep(
```

File: feature-runner/steps/device.ts [hunk 3]
```diff
@@ -21,190 +18,178 @@ import { getAttributesForDevice } from '../../devices/getAttributesForDevice.js'
...
-		log: {
-			step: { progress },
+			progress(`Device registered: ${fingerprint} (${id})`)
 		},
-	}: StepRunnerArgs<Record<string, any>>): Promise<StepRunResult> => {
-		const match = matchGroups(
-			Type.Object({
+	)
+
+const getDevice = ({
```
Comment (coderbyheart) [inline feature-runner/steps/device.ts]: Remove unused step definition.

File: features/HistoricalRequest.feature.md [hunk 4]
```diff
@@ -49,7 +56,7 @@ And I send this message via the websocket
   "message": "message",
   "payload": {
     "@context": "https://github.com/hello-nrfcloud/proto/historical-data-request",
-    "@id": "420eac59-5ce8-4751-b7d1-217811382095",
+    "@id": "${requestId}",
```
Comment (coderbyheart) [inline features/HistoricalRequest.feature.md]: Unrelated change. What's wrong with using a hardcoded ID here?

File: features/LastSeen.feature.md [hunk 5]
```diff
@@ -23,15 +31,12 @@ And the device `${fingerprint_deviceId}` publishes this message to the topic
...
-
 ## Retrieve last seen timestamp on connect
 
 Given I store `$fromMillis(${ts})` into `tsISO`
 
-When I connect to the websocket using fingerprint `${fingerprint}`
-
-<!-- @retry:tries=5,initialDelay=5000,delayFactor=1 -->
+When I connect to the websocket using fingerprint `${fingerprint}` in `5`
+seconds
```
Comment (coderbyheart) [inline features/LastSeen.feature.md]:
```suggestion
Given I wait for 5 seconds

When I connect to the websocket using fingerprint `${fingerprint}`
```

If user must wait 5 seconds before connect, add a separate wait step.

Comment (coderbyheart) [reply features/LastSeen.feature.md]: But I wonder, why must the user wait now?

File: package.json [hunk 6]
```diff
@@ -44,22 +44,22 @@
     "@aws-sdk/client-timestream-write": "3.410.0",
     "@aws-sdk/util-dynamodb": "3.410.0",
     "@nordicsemiconductor/asset-tracker-cloud-code-style": "12.0.86",
-    "@nordicsemiconductor/bdd-markdown": "5.6.12",
+    "@nordicsemiconductor/bdd-markdown": "6.0.3",
     "@nordicsemiconductor/cloudformation-helpers": "8.0.0",
     "@nordicsemiconductor/eslint-config-asset-tracker-cloud-typescript": "16.0.18",
-    "@nordicsemiconductor/firmware-ci-device-helpers": "14.0.198",
+    "@nordicsemiconductor/firmware-ci-device-helpers": "14.0.201",
```
Comment (coderbyheart) [inline package.json]: Unrelated change. Revert.

File: package.json [hunk 7]
```diff
@@ -44,22 +44,22 @@
...
     "@swc/cli": "0.1.62",
-    "@swc/core": "1.3.84",
+    "@swc/core": "1.3.83",
```
Comment (coderbyheart) [inline package.json]: Unrelated change. Revert.

File: package.json [hunk 8]
```diff
@@ -44,22 +44,22 @@
...
     "@swc/jest": "0.2.29",
-    "@types/aws-lambda": "8.10.119",
+    "@types/aws-lambda": "8.10.120",
```
Comment (coderbyheart) [inline package.json]: Unrelated change. Revert.

File: package.json [hunk 9]
```diff
@@ -44,22 +44,22 @@
...
     "@types/chai": "4.3.6",
     "@types/chai-subset": "1.3.3",
     "@types/glob": "8.1.0",
     "@types/jest": "29.5.4",
     "@types/lodash-es": "4.17.9",
     "@types/ws": "8.5.5",
     "@types/yazl": "2.4.2",
-    "@typescript-eslint/eslint-plugin": "6.6.0",
+    "@typescript-eslint/eslint-plugin": "6.7.0",
```
Comment (coderbyheart) [inline package.json]: Unrelated change. Revert.

File: package.json [hunk 10]
```diff
@@ -73,14 +73,14 @@
     "glob": "10.3.4",
     "husky": "8.0.3",
     "id128": "1.6.6",
-    "jest": "29.6.4",
+    "jest": "29.7.0",
     "jsonata": "2.0.3",
```
Comment (coderbyheart) [inline package.json]: Unrelated change. Revert.

File: package.json [hunk 11]
```diff
@@ -73,14 +73,14 @@
...
     "mqtt": "5.0.5",
     "p-retry": "6.0.0",
     "prettier": "3.0.3",
     "semver": "7.5.4",
     "tsmatchers": "4.0.12",
-    "tsx": "3.12.8",
+    "tsx": "3.12.9",
```
Comment (coderbyheart) [inline package.json]: Unrelated change. Revert.

File: package.json [hunk 12]
```diff
@@ -118,11 +118,11 @@
   "prettier": "@nordicsemiconductor/asset-tracker-cloud-code-style/.prettierrc",
   "dependencies": {
     "@aws-lambda-powertools/metrics": "1.12.1",
-    "@hello.nrfcloud.com/proto": "5.5.4",
+    "@hello.nrfcloud.com/proto": "5.5.10",
```
Comment (coderbyheart) [inline package.json:121]: Unrelated change. Revert.

File: package.json [hunk 13]
```diff
@@ -118,11 +118,11 @@
...
     "@middy/core": "4.6.1",
     "@nordicsemiconductor/from-env": "2.0.0",
     "@nordicsemiconductor/timestream-helpers": "5.0.0",
-    "@sinclair/typebox": "0.31.5",
+    "@sinclair/typebox": "0.31.14",
```
Comment (coderbyheart) [inline package.json:125]: Unrelated change. Revert.

**Review by coderbyheart — CHANGES_REQUESTED** [review]

File: feature-runner/run-features.ts (hunk 1 above)
Comment (pudkrong) [reply feature-runner/run-features.ts]: I just remove `chalk.magenta(info.context.title)` because `info.context` is not available since BDD markdown v.6

File: features/HistoricalRequest.feature.md (hunk 4 above)
Comment (pudkrong) [reply features/HistoricalRequest.feature.md]: When I debug the freaky test, I would like to have a unique request id. So, it is easier to search on the log.

File: features/LastSeen.feature.md (hunk 5 above)
Comment (pudkrong) [reply features/LastSeen.feature.md]: Last seen will be recorded in DB in async way. So, sometimes when you disconnect and reconnect immediately, the DB is not done yet and the test is failed.

File: package.json (hunk 12 above)
Comment (pudkrong) [reply package.json:121]: We need to upgrade `@hello.nrfcloud.com/proto` to use the same version of `@sinclair/typebox`

File: package.json (hunk 13 above)
Comment (pudkrong) [reply package.json:125]: We need the same version of `@sinclair/typebox` in `@nordicsemiconductor/bdd-markdown`

File: feature-runner/run-features.ts (hunk 1 above)
Comment (coderbyheart) [reply feature-runner/run-features.ts]: Ah, I see. `context` has been renamed to `step`, it now is only the instance of the Step.

File: feature-runner/steps/utils.ts
Comment (coderbyheart) [inline feature-runner/steps/utils.ts:1]: Remove.

File: features/RuntimeConfiguration.feature.md [hunk 14]
```diff
@@ -281,18 +287,16 @@ And there is this device shadow data for `${fingerprint_deviceId}` in nRF Cloud
...
-
 Soon the duration between 2 consecutive device shadow requests for
 `${fingerprint_deviceId}` should be `5` seconds
 
 ## Verify if changing fetching interval to 10 seconds, device shadow requests should be fired 1 time in 10 seconds
 
 Given device shadow fetching config for model `PCA20035+solar` is `10`
 
-When I connect to the websocket using fingerprint `${fingerprint}`
+When I wait for `5` seconds
```
Comment (coderbyheart) [inline features/RuntimeConfiguration.feature.md]: This is nothing the user should do, the 5 second delay should be handled in the following connect step. It's an implementation detail of the test system and should not be made part of the public API.

File: package.json (hunk 12 above)
Comment (coderbyheart) [inline package.json:121]: Unrelated change, revert.

File: package.json (hunk 13 above)
Comment (coderbyheart) [inline package.json:125]: Unrelated change, revert.

**Review by coderbyheart — CHANGES_REQUESTED** [review]

File: features/RuntimeConfiguration.feature.md (hunk 14 above)
Comment (pudkrong) [reply features/RuntimeConfiguration.feature.md]: But if we put the logic in web socket connection, it will apply for all even we need it for a specific case.

File: package.json (hunk 12 above)
Comment (pudkrong) [reply package.json:121]: This change is required to satisfy `tsc` because all `typebox` modules must be the same version.

File: package.json (hunk 13 above)
Comment (pudkrong) [reply package.json:125]: This change is required to satisfy tsc because all typebox modules must be the same version.

File: features/RuntimeConfiguration.feature.md (hunk 14 above)
Comment (coderbyheart) [reply features/RuntimeConfiguration.feature.md]: Yes, it's the DB that is lacking behind so this needs an internal retry for the connection db.

Comment (coderbyheart) [reply features/RuntimeConfiguration.feature.md]: And this check should be in the promise of the step that **provides** the data in the db. So this step waits until it is available and only then do the next steps execute. If the step is later called, the promise is already fulfilled and it will continue immediately.

File: feature-runner/run-features.ts (hunk 1 above)
Comment (coderbyheart) [inline feature-runner/run-features.ts]:
```suggestion
			console.error(
				ts(),
				chalk.magenta.dim(info.step.keyword),
				chalk.magenta(info.step.title),
				...args.map((arg) => chalk.cyan(print(arg))),
			),
```
